### PR TITLE
Clean up (new images, text, variable names)

### PR DESCRIPTION
**Pull request by random-forests**
I really like this tutorial. Went through and did a round of cleanup (if you like these images we can host elsewhere, they're on my personal repo for now - we own them). 

P.S., while cleaning up I noticed this code is running suuuuuper slow (like 15 mins an epoch). That problem exists in the current version, so figured I'd get these updates in regardless.

Review NB: https://app.reviewnb.com/tensorflow/docs/pull/332/files/
Colab: https://colab.sandbox.google.com/github/random-forests/docs/blob/master/site/en/r2/tutorials/generative/dcgan.ipynb

### CODE REVIEW

**Comment by yashk2810** [comment]
I'll take a look at this tomorrow 😀

**Comment by yashk2810** [comment]
Did you try installing the gpu version. (pip install tf-nightly-gpu-2.0-preview)

Its taking 33 secs for me.

File: site/en/r2/tutorials/generative/dcgan.ipynb [hunk 1]
```diff
@@ -1,143 +1,197 @@
 {
```
Comment (MarkDaoust) [inline site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Use tfds?</p><p><br></p><p>(This adds a 30s delay,  🙁)</p><p><br></p><p>```</p><p>!pip install tfds_nightly</p><p>import tensorflow_datasets as tfds</p><p><br></p><p>train_dataset = tfds.load('mnist', split = tfds.Split.TRAIN)</p><p><br></p><p>def format_dataset(features):</p><p>return (features['image']-127.5)/127.5)</p><p><br></p><p>train_dataset = mnist.map(format_dataset)</p><p>```</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (MarkDaoust) [inline site/en/r2/tutorials/generative/dcgan.ipynb]: <p>"Except the output layer which uses"</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (MarkDaoust) [inline site/en/r2/tutorials/generative/dcgan.ipynb]: <p>This chunk would be easier to read if we `import tensorflow.keras.layers as layers`, then `model.add(layers.Dense(...))`</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (MarkDaoust) [inline site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Prefer imperative to "lets":</p><p><br></p><p>"Use the (as yet ..."</p><p><br></p><p>More generally, try to avoid "we" and "lets": https://developers.google.com/style/person</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (MarkDaoust) [inline site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Maybe add a comment like: </p><p>"The model output is a binary-logit. It will be trained to output positive values for real images and negative values for fake images. The current output is near zero, indicating uncertainty, which is what you expect for an untrained model.  </p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (MarkDaoust) [inline site/en/r2/tutorials/generative/dcgan.ipynb]: <p>This text isn't totally accurate given that the model is outputting logits.</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Comment (MarkDaoust) [inline site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Typo: "updat" --&gt; "update" </p><p><br></p><p>Maybe: "So we don't update the batchnorm parameters" --&gt; "So all layers run in inference mode (batchnorm)."</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='41'/>

Comment (yashk2810) [reply site/en/r2/tutorials/generative/dcgan.ipynb]: cast to float32 too.

30 secs delay per epoch?

Comment (yashk2810) [reply site/en/r2/tutorials/generative/dcgan.ipynb]: I am seeing a 3 sec delay.

**Comment by random-forests** [comment]
> Did you try installing the gpu version. (pip install tf-nightly-gpu-2.0-preview)

You rock, that fixed it. 

Thanks for other comments! Will do another pass this evening.

P.S., tried ReviewNB - pretty cool.

File: site/en/r2/tutorials/generative/dcgan.ipynb (hunk 1 above)
Comment (random-forests) [reply site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Thanks, fixed.</p>

Comment (random-forests) [reply site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Agreed &amp; updated, thanks.</p>

Comment (random-forests) [reply site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Cool, made some updates.</p>

Comment (random-forests) [reply site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Cool, done.</p>

Comment (random-forests) [reply site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Updated &amp; shortened, thanks</p>

Comment (random-forests) [reply site/en/r2/tutorials/generative/dcgan.ipynb]: <p>Done</p>

**Comment by random-forests** [comment]
Thanks for the comments! Uploaded v2.

**Comment by yashk2810** [comment]
Use tfds?

File: site/en/r2/tutorials/generative/dcgan.ipynb [hunk 2]
```diff
@@ -1,668 +1,693 @@
...
-        "id": "YfIk2es3hJEd"
+        "colab": {}
       },
-      "outputs": [],
+      "cell_type": "code",
       "source": [
-        "from __future__ import absolute_import, division, print_function\n",
-        "\n",
-        "!pip install tf-nightly-2.0-preview\n",
         "import tensorflow as tf\n",
```
Comment (yashk2810) [inline site/en/r2/tutorials/generative/dcgan.ipynb]: import tensorflow as tf #gpu

Comment (yashk2810) [reply site/en/r2/tutorials/generative/dcgan.ipynb]: this for internal.

Sorry

**Comment by random-forests** [comment]
> Use tfds?

Noticed a slowdown w/ tfds as well. Filed b/124258618

**Comment by yashk2810** [comment]
> slowdown w/ tfds as well

Slowdown where?

For training, you get afaik 1 sec delay because of the shuffle buffer filing up again.

For loading, it takes some time because it first writes raw files to TFRecord and then loads it back from there.

**Comment by random-forests** [comment]
> Slowdown where?

In the training loop, the per epoch time increases when using tfds (in highly scientific testing... running a couple of times in Colab) =D The latest was significant (more than a couple seconds added / epoch).

**Comment by random-forests** [comment]
Okay! The slowdown is a results of tfds loading the data from disk. I'm sure there's a way to do an in-memory solution, but until we figure that out might as well and merge / close this out if otherwise looks good, can always circle back and swap out the dataset loader later.

File: site/en/r2/tutorials/generative/dcgan.ipynb [hunk 3]
```diff
@@ -1,668 +1,693 @@
...
+        "    with tf.GradientTape() as gen_tape, tf.GradientTape() as disc_tape:\n",
+        "      generated_images = generator(noise, training=True)\n",
+        "\n",
+        "      real_output = discriminator(images, training=True)\n",
+        "      fake_output = discriminator(generated_images, training=True)\n",
+        "\n",
+        "      gen_loss = generator_loss(fake_output)\n",
+        "      disc_loss = discriminator_loss(real_output, fake_output)\n",
+        "\n",
+        "    gradients_of_generator = gen_tape.gradient(gen_loss, generator.variables)\n",
```
Comment (yashk2810) [inline site/en/r2/tutorials/generative/dcgan.ipynb]: trainable_variables

Here and below too.

**Comment by yashk2810** [comment]
@random-forests Final change then we can pull this in :)

**Comment by random-forests** [comment]
Re: trainable_variables, good catch, thank you! Fixed.

Also, silly CLA bot :)

**Comment by lamberta** [comment]
http://signcla/random-forests

(Looks like you're registered with your gmail which might account for the CLA bot wackiness here.)

**Comment by lamberta** [comment]
@random-forests Mark has some comments on this here: cl/234740506

**Comment by random-forests** [comment]
Cool, responded to comments on cl/234740506 here (added images, did the rename), thanks! Re: ownership, the images belong to Google.
